src/magellan_vision/scripts/line_detector.py

```diff
@@ -1,211 +1,142 @@
-    # final = cv2.addWeighted(processed, 0.8, linesAndCircles, 0.5,0)
-    # cv2.imshow('Original', frame)
-    # cv2.imshow('Combo', final)
-    # print(skipped_frames)
-
-    # FIXME: what is this crap, please FIX ME
-    # plt.plot(np.unique(x_array_right),
-    #          np.poly1d(np.polyfit(x_array_right, y_array_right, 1))
-    #          (np.unique(x_array_right)))
-
-    # right_line_slope,right_line_intercept = best_fit(np.asarray(x_array_right),np.asarray(y_array_right))
-    # left_line_slope,left_line_intercept = best_fit(np.asarray(x_array_left),np.asarray(y_array_left))
-
-    toc = time.time()
-    delta_t = toc - tic
-    print("FPS: {}".format(1/delta_t))
-
-    key = cv2.waitKey(25)
-    if key == 27:
-        break
-
-video.release()
-cv2.destroyAllWindows
+                cv2.line(line_img, (x1, y1), (x2, y2), color, thickness)
+        img = cv2.addWeighted(img, 0.8, line_img, 1.0, 0.0)
+        return img
+
+    def best_fit(x_points, y_points):
+
+        m = (((mean(x_points)*mean(y_points)) - mean(x_points*y_points)) /
+             ((mean(x_points)*mean(x_points))-mean(x_points*x_points)))
+        b = mean(y_points) - m*mean(x_points)
+
+        return m, b
+
+    def detect(self, cv_image):
+
+        gray = cv2.cvtColor(cv_image, cv2.COLOR_BGR2GRAY)
+        blurred = cv2.GaussianBlur(gray, (3, 3), 0)
+        ret, thresh4 = cv2.threshold(blurred, 200, 255, cv2.THRESH_TOZERO)
+        size = np.size(thresh4)
+        skel = np.zeros(thresh4.shape, np.uint8)
+
+        skeleton = self.skeletize(thresh4, size, skel)
+
+        dst = cv2.Laplacian(skeleton, ddepth, ksize=kernel_size)
+        abs_dst = cv2.convertScaleAbs(dst)
+
+        linesP = cv2.HoughLinesP(abs_dst, rho=6, theta=np.pi / 60, threshold=50,
+                                 lines=np.array([]), minLineLength=30, maxLineGap=10)
+
+        if linesP is not None:
+            for line in linesP:
+                for x1, y1, x2, y2 in line:
+                    slope = (y2 - y1) / (x2 - x1)
+                    # <-- Calculating the slope.
+                    if math.fabs(slope) < .5:
+                        # <-- Only consider extreme slope
+                        continue
+                    if x1 < cv_image.shape[1]/2 and x2 < cv_image.shape[1]/2:
+                        # <-- If the slope is negative, left group
+                        cv2.circle(cv_image, (x1, y1), (5), (0, 0, 255), 3)
+                        cv2.circle(cv_image, (x2, y2), (5), (0, 0, 255), 3)
+                    else:  # <-- Otherwise, right group.
+                        cv2.circle(cv_image, (x1, y1), (5), (0, 255, 0), 3)
+                        cv2.circle(cv_image, (x2, y2), (5), (0, 255, 0), 3)
+
+        cv2.namedWindow('Detections', cv2.WINDOW_AUTOSIZE)
+        cv2.imshow('Detections', cv_image)
+        cv2.waitKey(25)
+
+    def clean_up(self):
+        cv2.destroyAllWindows()
+
+
+def main():
+    pipeline = rs.pipeline()
```

Comment:
With the ROS version, we would be using realsense ROS instead of streaming direct from the camera. If we do it this way, only one process can use the camera images.